actors/evm/tests/contracts/TransientStorageTest.sol

```diff
@@ -0,0 +1,164 @@
+        testBasicFunctionality();
+        testLifecycleValidation();
+    }
+
+    // Test 1: Basic Functionality
+    function testBasicFunctionality() public {
+        uint256 slot = 1;
+        uint256 value = 42;
+
+        // Store value using TSTORE
+        assembly {
+            tstore(slot, value)
+        }
+
+        // Retrieve value using TLOAD
+        uint256 retrievedValue;
+        assembly {
+            retrievedValue := tload(slot)
+        }
+
+        require(retrievedValue == value, "TLOAD did not retrieve the correct value");
+
+        // Verify TLOAD from uninitialized location
+        uint256 uninitializedSlot = 2;
+        uint256 uninitializedValue;
+        assembly {
+            uninitializedValue := tload(uninitializedSlot)
+        }
+
+        require(uninitializedValue == 0, "Uninitialized TLOAD did not return zero");
+    }
+
+    // Test 2.1: Verify transient storage clears after transaction
+    function testLifecycleValidation() public {
+        uint256 slot = 3;
+        uint256 value = 99;
+
+        // Store value using TSTORE
+        assembly {
+            tstore(slot, value)
+        }
+
+        // Verify it exists within the same transaction
+        uint256 retrievedValue;
+        assembly {
+            retrievedValue := tload(slot)
+        }
+        require(retrievedValue == value, "TLOAD did not retrieve stored value within transaction");
+    }
+
+    function testLifecycleValidationSubsequentTransaction() public {
+        // Test clearing by re-calling as a new transaction
+        uint256 slot = 3;
+        bool cleared = isStorageCleared(slot);
+        require(cleared, "Transient storage was not cleared after transaction");
+    }
+
+    function isStorageCleared(uint256 slot) public view returns (bool) {
+        uint256 retrievedValue;
+        assembly {
+            retrievedValue := tload(slot)
+        }
+        return retrievedValue == 0; // True if cleared
+    }
+
+    // Test 2.2: Verify nested contract independence
+    function testNestedContracts(address other) public returns (bool) {
+
+        uint256 slot = 4;
+        uint256 value = 88;
+
+        TransientStorageTest nested = TransientStorageTest(other);
+
+        // Store in this contract's transient storage
+        assembly {
+            tstore(slot, value)
+        }
+
+        // Call nested contract to write its own transient storage
+        nested.writeTransientData(4, 123);
+
+        // Verify this contract's data is unchanged
+        uint256 retrievedValue;
+        assembly {
+            retrievedValue := tload(slot)
+        }
+        require(retrievedValue == value, "Nested contract interfered with this contract's storage");
+
+        // Verify nested contract's data independently
+        uint256 nestedValue = nested.readTransientData(4);
+        require(nestedValue == 123, "Nested contract data incorrect");
+        return true;
+    }
+
+        // New function to test reentry scenario
```

Comment:
```suggestion
    // New function to test reentry scenario
```

Reply:
Ah yeah I had to refactor these a few times and lost the numbering  fun also it could use an auto fmt